DEVELOPMENTLOG.md

```diff
@@ -1,4 +1,6 @@
-# OpenLoco version 24.01+ (???)
+# OpenLoco version 24.01.1 (2024-01-16)
+
+In 24.01 we accidentally introduced a bug that causes some crashes when loading save games on Windows machines. This one was tricky to spot in testing since it only occured on Windows, and only in release builds, which are not often used when testing in development. This is fixed in 24.01.1.
```

Comment:
Could you line-wrap this paragraph? Might also want to include that it involves Safe Structured Exception Handling. LGTM otherwise.